### PR TITLE
Static mode, server, building, UI/UX

### PR DESCRIPTION
**Pull request by yaskevich**
- Annotatrix in static mode (when ran from file, without server)
  - saves to a LocalStorage, #339 
  - doesn't try to communicate with a server, #353   
- Minor fixes (building on Windows, improvements in a build script, etc.)
- Small UI improvement: favicon, it was built with [faviator](https://www.npmjs.com/package/faviator), #355

### CODE REVIEW

File: client/app.js [hunk 1]
```diff
@@ -76,16 +93,17 @@ class App {
 
     // serialize the corpus
     let serial = this.corpus.serialize();
-
+    console.log("this.corpus.serialize", serial);
```
Comment (jonorthwash) [inline client/app.js]: Is this needed, or was this to help you figure out the issue?

File: client/undo-manager.js [hunk 2]
```diff
@@ -59,8 +60,9 @@ class UndoManager {
       return false;
 
     // do some comparisons here to change for changes
-    console.log(serial)
-    console.log(this.current)
+    console.log("serial", serial) // updated
+    utils.storage.backup(serial);
+    console.log("current", this.current)
```
Comment (jonorthwash) [inline client/undo-manager.js]: Are these logs important, or are they just for debugging?

Comment (yaskevich) [reply client/undo-manager.js]: They are for debugging, but they have not to be dropped manually, considering that there are much more of them in a code base. And actually I need them in dev environment. So please accept unchanged, I provide solution for your concern in next PR, closing issue #358.

File: server/public/html/index.html [hunk 3]
```diff
@@ -1,43 +1,239 @@
...
 <head>
 	<meta http-equiv="content-type" content="text/html; charset=utf-8" />
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<link rel="icon" href="../favicon.png" sizes="230x230" type="image/png">
   <title>UD Annotatrix</title>
 	<link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css" />
-	<link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css" />
+	<!-- <link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css" /> -->
 	<link rel="stylesheet" type="text/css" href="../css/style.css" />
+	<style>
```
Comment (jonorthwash) [inline server/public/html/index.html]: Why are you including style this way?  Can't this just be in the css file?

File: server/public/html/index.html [hunk 4]
```diff
@@ -1,43 +1,239 @@
...
+     body {
+       padding-top: 100px;
+     }
+		 .actmode{
```
Comment (jonorthwash) [inline server/public/html/index.html]: Why the change in indentation?  Are you using spaces or tabs?

File: server/public/html/index.html [hunk 5]
```diff
@@ -1,43 +1,239 @@
...
+		 	line-height: .5rem;
+		 }
+  </style>
+	<script src="../js/jquery.min.js"></script>
+	<script src="../js/bootstrap.bundle.js"></script>
+	<script type="text/javascript">
```
Comment (jonorthwash) [inline server/public/html/index.html]: Why are you embedding javascript directly, and not in the libraries?

File: server/public/html/index.html [hunk 6]
```diff
@@ -1,43 +1,239 @@
...
+									 <button id="upload-to-server" class="btn btn-outline-secondary" type="button">Upload</button>
+	              </div>
+	              <div class="custom-file">
+	                <input type="file" class="custom-file-input" id="fileupload"
+	                  aria-describedby="fileupload">
+	                <label class="custom-file-label" for="inputGroupFile01">Select local file</label>
+	              </div>
+	            </div>
+		  </li>
+		  <!-- <li class="nav-item dropdown">
```
Comment (jonorthwash) [inline server/public/html/index.html]: What is this for?

File: server/public/html/index.html [hunk 7]
```diff
@@ -1,43 +1,239 @@
...
+					<!-- <form class="form-inline my-2 my-lg-0">
+					<input class="form-control mr-sm-2" type="search" placeholder="URL" aria-label="Search">
+					<button class="btn btn-outline-success my-2 my-sm-0" type="submit">Load</button>
+				</form> -->
+			 </li>
+		  <li class="nav-item">
+				<div class="input-group">
+	              <div class="input-group-prepend">
+	                <!-- <span class="input-group-text" id="inputGroupFileAddon01">Upload</span> -->
+									 <button id="upload-to-server" class="btn btn-outline-secondary" type="button">Upload</button>
```
Comment (jonorthwash) [inline server/public/html/index.html]: Why the extra indent?

File: server/views/help.ejs [hunk 8]
```diff
@@ -29,55 +29,55 @@
     </tr>
     <tr>
       <td>Delete a deprel</td>
-      <td>Select all the deprels to be removed with right click, then press either <b>del</b> or <b>backspace</b>. If you want to unselect all the arcs you've just selected, press <b>esc</b>.</td>
+      <td>Select all the deprels to be removed with right click, then press either <kbd>del</kbd> or <kbd>backspace</kbd>. If you want to unselect all the arcs you've just selected, press <kbd>esc</kbd>.</td>
```
Comment (jonorthwash) [inline server/views/help.ejs]: Is `kbd` standard html?  Wouldn't it be better to do something like `<span class="kbd">...</span>`?

File: server/views/index.ejs [hunk 9]
```diff
@@ -1,89 +1,292 @@
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```
Comment (jonorthwash) [inline server/views/index.ejs]: Why did you remove this?

File: server/views/settings.ejs [hunk 10]
```diff
@@ -1,5 +1,5 @@
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```
Comment (jonorthwash) [inline server/views/settings.ejs]: Why did you remove this?

File: server/views/settings.ejs [hunk 11]
```diff
@@ -1,5 +1,5 @@
...
-<html xmlns="http://www.w3.org/1999/xhtml">
```
Comment (jonorthwash) [inline server/views/settings.ejs]: And this?

File: server/public/html/index.html (hunk 3 above)
Comment (yaskevich) [reply server/public/html/index.html]: Because I needed some styles to be specific for a main page.

File: server/public/html/index.html (hunk 4 above)
Comment (yaskevich) [reply server/public/html/index.html]: I use Atom, and it seems to use tabs by default, I switch it to spaces.

File: server/public/html/index.html (hunk 5 above)
Comment (yaskevich) [reply server/public/html/index.html]: Because code of index page and for annotation page is different.

File: server/public/html/index.html (hunk 6 above)
Comment (yaskevich) [reply server/public/html/index.html]: Removed.

File: server/public/html/index.html (hunk 7 above)
Comment (yaskevich) [reply server/public/html/index.html]: I will reindent it with Atom Beautify plugin.

File: server/views/index.ejs (hunk 9 above)
Comment (yaskevich) [reply server/views/index.ejs]: It has `<!DOCTYPE html>`.

File: server/views/settings.ejs (hunk 10 above)
Comment (yaskevich) [reply server/views/settings.ejs]: It has `<!DOCTYPE html>`.

File: server/views/settings.ejs (hunk 11 above)
Comment (yaskevich) [reply server/views/settings.ejs]: It has `<!DOCTYPE html>` and `html` tag. If we use Bootstrap and stuff we are surely with HTML5, so it's better to have headers according to HML5 standards, not HTML4 ones.

**Comment by yaskevich** [comment]
Currently, it's a new set of changes, I didn't take into account that previous set was not merged yet. So maybe it would be better to revert last commit, to wait for previous set with its PR would be accepted and then re-commit?

File: server/views/help.ejs (hunk 8 above)
Comment (yaskevich) [reply server/views/help.ejs]: it's included in Bootstrap framework and works.

**Comment by jonorthwash** [comment]
Really you should have committed the additional stuff as separate PR.  But at this point I don't think it makes sense to revert the commit.  Could you fix the indentation and spelling issues (and anything else you said you would), and then I'll merge the PR.

Also, it looks like your branch is missing a couple commits from this one?

File: server/public/html/index.html [hunk 12]
```diff
@@ -1,239 +1,123 @@
...
+    <!-- end snippet -->
+    <div class="container">
+        <div id="welcome-content">
+
+            <!-- <h1 id="welcome-banner">Welcome to UD Annotatrix</h1> -->
+
+            <!-- <form method="post" enctype="multipart/form-data" action="./upload">
+        <input type="file" name="file" />
+        <input type="submit" value="Upload to server" />
+    </form> -->
```
Comment (jonorthwash) [inline server/public/html/index.html]: What happened to the formatting of this file?  Especially these commented-out blocks?

It was better with consistent tabs than your reformatting with spaces.  (Personally I prefer tabs.)

Could you revert the state of this file if it's just reformatting?

**Comment by jonorthwash** [comment]
@yaskevich, could you revert the reformatting of `index.html`?  And stop adding to this PR—open new ones for additional features.

File: server/public/html/index.html (hunk 12 above)
Comment (yaskevich) [reply server/public/html/index.html]: I autoformatted everything in code editor. Actually, I deal with ejs only, html is autogenerated.
It's right,commented blocks were not properly indented. All I can do now is to reformat them in ejs, so in next commits they would be indented.

**Comment by maryszmary** [comment]
@jonorthwash, what should be finished for pr to be merged?

**Comment by yaskevich** [comment]
> @yaskevich, could you revert the reformatting of `index.html`?

Unfortunately, I added some changes in subsequent commits to `index.ejs`, so now it is too hard to reconstruct original indentation.

**Comment by jonorthwash** [comment]
> Unfortunately, I added some changes in subsequent commits to `index.ejs`

This is exactly why it's a bad idea to make formatting changes and substantive changes in the same commit.  Please revert the commit and figure out what you did.  Also, I asked about `index.html`, not `index.ejs`...

**Comment by jonorthwash** [comment]
Can't you just mass-replace spaces with tabs?

**Comment by yaskevich** [comment]
The reason why formating changed: 
https://github.com/jonorthwash/ud-annotatrix/pull/356#discussion_r295577729

Why I mentioned ejs
https://github.com/jonorthwash/ud-annotatrix/pull/356#discussion_r298281610

> Can't you just mass-replace spaces with tabs?

At the beginning of the work on the project I was wondering whether the formatting is consistent in the whole project. After I got familiar with a whole codebase I could claim that it does not (sometimes there are tabs, sometimes there are spaces). Unification would drastically change all the code base (in diffs).

If change of the formatting caused a problem for making diff and therefore for code review, (as a solution) I could pretty-print any file and its earlier version with the same prettifier.

I understand that it is great when every file of the project follows the same formatting rules. Unfortunately, JavaScript (the more so HTML) is not Python, and if someone prefers one style, and someone does another, the browser/interpreter doesn't throw any errors, so it leads to some non-conformity. Also, maybe, sometimes I change formatting because in such way it is more clear for me and it helps with adding new code.

Actually, I have noticed some inconsistencies and withdrawal of common practices in the code base that no one has mentioned yet. But considering that the project has quite specific tasks to deal with (interaction with Github API or debugging tricky issues which rooted in `notatrix`), it seems that it would be better to focus on the new code, although I would be happy to make the whole code aesthetically ideal.

**Comment by yaskevich** [comment]
If it is about some specific changes (as converting all the spaces of index.ejs, therefore index.html to tabs), I can easily do that, especially if it could go with next PR.

**Comment by jonorthwash** [comment]
It's okay to fix spaces and tabs in a file en masse to unify formatting, or to fix formatting here and there to make things easier for you to work with.  What I would really like to see is a commit that shows what else you changed.  You could revert the commit, manually restore the formatting in the newer version (replace any 4 spaces with tab throughout the file), diff it with the file before the commit, commit that patch, and then fix the formatting in another commit.

**Comment by yaskevich** [comment]
I've changed these two files (`index.ejs` and `index.html`), every 4 spaces were replaced with 1 tab.

**Comment by jonorthwash** [comment]
> What I would really like to see is a commit that shows what else you changed.

Could you at least paste a diff of the new `index.ejs` and the one before your changes, so I can see what else was changed?

**Comment by yaskevich** [comment]
I pretty-printed both `index.ejs` from the original repo and `index.ejs` from my repo with [HTML Formatter](https://htmlformatter.com/): result is [here](https://gist.github.com/yaskevich/a677aabcc639617af8c71ed1f91f99cf/revisions?diff=unified). 

Actually, index page, despite it remained functionally ans structurally the same, changed its HTML markup, there was bare HTML initially, and in new version Bootstrap classes are used, which makes the page adaptive.

So, this view
2019-07-23 18_00_47-UD Annotatrix

turned into that:

2019-07-23 18_02_12-UD Annotatrix
